### PR TITLE
Feature/phone

### PR DESCRIPTION
**Pull request by shekeraoleksandr**
What I did:
- Phone number generation
- Imei generation 
- Modified the Helper class with methods that were needed to implement the Phone class
- Created Luhn check class
- Wrote tests for all of this 
- Also added some phone numbers templates for some countries

### CODE REVIEW

File: include/faker-cxx/Helper.h [hunk 1]
```diff
@@ -55,6 +59,173 @@ class Helper
...
+     * @return The string replaced symbols with digits.
+     *
+     * @code
+     * Helper::replaceSymbolWithNumber() // ""
+     * Helper::replaceSymbolWithNumber("#####") // "04812"
+     * Helper::replaceSymbolWithNumber("!####") // "27378"
+     * Helper::replaceSymbolWithNumber("Your pin is: !####") // "29841"
+     * @endcode
+     */
+    static std::string replaceSymbolWithNumber(std::string str, const char & symbol = '#')
```
Comment (cieslarmichal) [inline include/faker-cxx/Helper.h:79]: please move implementations in source file

File: include/faker-cxx/Helper.h [hunk 2]
```diff
@@ -55,6 +59,173 @@ class Helper
...
+            string =
+                string.substr(0, static_cast<unsigned long>(token.position())) +
+                std::to_string(Number::integer(min, max)) +
+                string.substr(static_cast<unsigned long>(token.position() + token.length()));
+        }
+
+        return string;
+    }
+
+//    /**
```
Comment (cieslarmichal) [inline include/faker-cxx/Helper.h:199]: remove please

File: include/faker-cxx/LuhnCheck.h [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    /**
+     * @brief Returns luhn checksum value for the given value.
+     *
+     * @param inputString The string to calculate the check value for.
+     *
+     * @return The luhn checksum value for the given value.
+     *
+     */
+    static int luhnCheckSum(const std::string& inputString) {
```
Comment (cieslarmichal) [inline include/faker-cxx/LuhnCheck.h:20]: please move it to src/common (its not the faker api) and create LuhnCheck.cpp file with implementations

File: include/faker-cxx/Phone.h [hunk 4]
```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <optional>
+#include <string>
+
+namespace faker
+{
+class Phone
```
Comment (cieslarmichal) [inline include/faker-cxx/Phone.h:8]: please add a second number(PhoneNumberCountryFormat format) method with PhoneNumberCountryFormat enum with values like PhoneNumberCountryFormat::Italy, and with implementation add mapping from enum to concrete country formats from PhoneNumbers.h (you can check my code in Location module I did the setup for addresses).

**Review by cieslarmichal — CHANGES_REQUESTED** [review]
Good job, I added couple comments.

Also: add an <algorithm> header to PhoneTest.cpp because windows build failed.

D:\a\faker-cxx\faker-cxx\src\modules\phone\PhoneTest.cpp(17,21): error C2039: 'all_of': is not a member of 'std'

File: include/faker-cxx/Helper.h (hunk 1 above)
Comment (shekeraoleksandr) [reply include/faker-cxx/Helper.h]: Done

File: include/faker-cxx/Helper.h (hunk 2 above)
Comment (shekeraoleksandr) [reply include/faker-cxx/Helper.h]: Done

File: include/faker-cxx/LuhnCheck.h (hunk 3 above)
Comment (shekeraoleksandr) [reply include/faker-cxx/LuhnCheck.h]: Done

File: src/modules/phone/Phone.cpp
Comment (shekeraoleksandr) [inline src/modules/phone/Phone.cpp:1]: Have a look and tell me if this is what you wanted to see

File: src/modules/phone/Phone.cpp [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    std::map<PhoneNumberCountryFormat, std::string> formatMap;
+
+    // Loop through all the PhoneNumberCountryFormat enum values
+    for (auto i = static_cast<unsigned long>(PhoneNumberCountryFormat::Default); i <= static_cast<unsigned long>(PhoneNumberCountryFormat::Zimbabwe); i++) {
+        auto formatEnum = static_cast<PhoneNumberCountryFormat>(i);
+        formatMap[formatEnum] = phoneNumbers[i];
+    }
+
+    return formatMap;
+}
```
Comment (shekeraoleksandr) [inline src/modules/phone/Phone.cpp:53]: Have a look and tell me if this is what you wanted to see

**Review by cieslarmichal — APPROVED** [review]
Nice! That is precisely what I meant :) 

Please join out community in discord channel: https://discord.com/invite/h2ur8H6mK6

**Pull request merged**